SU-HSDO/suhumsci
# Fix(STN-931): Fix incorrect padding for raised horizontal cards

# READY FOR REVIEW

## Summary
Date Stack Horizontal Card without the date tiles (which is our default view for past events) and no imagery - does not have enough padding. This padding is only an issue on Colorful.

This work alters our CSS styling to only remove the padding on the left from horizontal cards using, raised cards, no image, and without the date tile in the cards. I maintained the padding removal on the left with a date tile stack because that seemed to be the intentional styling, without H&S in the office I chose this path. We can remove the padding in that type of instance as well later.

Jira Card: https://sparkbox.atlassian.net/browse/STN-931

QA page: https://sparkbox-sandbox-stage.stanford.edu/date-stack-horizontal-card-no-image

## Need Review By (Date)
12/21/21

## Urgency
medium

## Steps to Test
1. npm run test
2. If you've synced with staging recently this page should be available for you to test with: https://sparkbox-sandbox.suhumsci.loc/date-stack-horizontal-card-no-image
3. If that page is unavailable to you, you can add and adjust a duplicate View manually.
- You will use this page to view the work: https://sparkbox-sandbox.suhumsci.loc/qa/date-stacked-horizontal-card
- Go to the Econ Events View: https://sparkbox-sandbox.suhumsci.loc/admin/structure/views/view/custm_custom_econ_events
- Click on "Date Stacked Horizontal" Display.
![image](https://user-images.githubusercontent.com/9355698/146848065-55a3c771-414f-412b-8afe-146cde53a88e.png)
- Duplicate the View, using the drop-down of the button on the right of the Displays area.
![image](https://user-images.githubusercontent.com/9355698/146848146-d6ef8384-2e39-4c63-b825-5c458799caef.png)
- After you've duplicated the View, change the Display Name of that view to be different from the view you just duplicated to `Date Stacked Horizontal - no images` under "Display Name".
![image](https://user-images.githubusercontent.com/9355698/146848274-d39a77ba-92ec-4a1a-9aa7-a7aeedf39981.png)
- Change the Pattern Display under (Format > Show > Settings).
![image](https://user-images.githubusercontent.com/9355698/146848519-9cc5911e-4b1e-4a67-ac0d-141574ed40ce.png)
- This will allow you to change the display of the Image and Date fields to Hidden within the Pattern.
![image](https://user-images.githubusercontent.com/9355698/146848467-8e1f2231-1d42-4be9-9888-ce853345267f.png)
- Click "Save" at the bottom of the View page. (If you miss this step, the changes won't apply globally).
4. Return to https://sparkbox-sandbox.suhumsci.loc/qa/date-stacked-horizontal-card, and click "Edit" to edit this page.
5. Click on the Paragraph for the "View" and edit that View option, change the dropdown for the Display to your new "no image" option.
![image](https://user-images.githubusercontent.com/9355698/146848721-e1fc9074-aa86-4926-b7af-f00f5ca50343.png)
6. Save the page, you should now have a page with the same display as the QA site.
7. You could also duplicate that first view and change the second one, so you have two views on the page to compare for visual regression.

## Verification items
- [x] Padding is correct on Colorful theme for these cards with `hb-raised-cards` applied.

These components remain unaffected:
- [x] Adding `hb-raised-cards--uniform-height` doesn't affect the padding.
- [x] Date Stack Horizontal Card with an image (without raised cards) [Colorful]
- [x] Date Stack Horizontal Card without an image (without raised cards) [Colorful]
- [x] Date Stack Horizontal Card with an image using .hb-raised-card [Colorful]
- [x] Date Stack Horizontal Card with an image on Traditional.
- [x] Date Stack Horizontal Card without an image on Traditional.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
